flickit-assessment-core/src/main/java/org/flickit/assessment/core/adapter/out/calculate/ConfidenceLevelCalculateInfoLoadAdapter.java

```diff
@@ -0,0 +1,199 @@
+
+    private final SubjectRestAdapter subjectRestAdapter;
+    private final QuestionRestAdapter questionRestAdapter;
+
+    record Context(List<QuestionDto> allQuestionsDto,
+                   List<AnswerJpaEntity> allAnswerEntities,
+                   List<QualityAttributeValueJpaEntity> allQualityAttributeValueEntities,
+                   List<SubjectValueJpaEntity> subjectValueEntities,
+                   Map<Long, SubjectDto> subjectIdToDto,
+                   Map<Long, Integer> qaIdToWeightMap) {
+    }
+
+    @Override
+    public AssessmentResult load(UUID assessmentId) {
+        AssessmentResultJpaEntity assessmentResultEntity = assessmentResultRepo.findFirstByAssessment_IdOrderByLastModificationTimeDesc(assessmentId)
+            .orElseThrow(() -> new ResourceNotFoundException(CALCULATE_ASSESSMENT_ASSESSMENT_RESULT_NOT_FOUND));
+        UUID assessmentResultId = assessmentResultEntity.getId();
+        Long assessmentKitId = assessmentResultEntity.getAssessment().getAssessmentKitId();
+
+        /*
+         load all subjectValue and attributeValue entities
+         that are already saved with this assessmentResult
+         */
+        List<SubjectValueJpaEntity> subjectValueEntities = subjectValueRepo.findByAssessmentResultId(assessmentResultId);
+        List<QualityAttributeValueJpaEntity> allQualityAttributeValueEntities = qualityAttrValueRepo.findByAssessmentResultId(assessmentResultId);
+
+        /*
+        load all subjects and their related attributes (by assessmentKit)
+        and create some useful utility maps
+        */
+        List<SubjectDto> subjectsDto = subjectRestAdapter.loadSubjectsDtoByAssessmentKitId(assessmentKitId);
+        Map<Long, Integer> qaIdToWeightMap = subjectsDto.stream()
+            .flatMap(x -> x.qualityAttributes().stream())
+            .collect(toMap(QualityAttributeDto::id, QualityAttributeDto::weight));
+        Map<Long, SubjectDto> subjectIdToDto = subjectsDto.stream()
+            .collect(toMap(SubjectDto::id, x -> x));
+
+        // load all questions with their impacts (by assessmentKit)
+        List<QuestionDto> allQuestionsDto = questionRestAdapter.loadByAssessmentKitId(assessmentKitId);
+
+        // load all answers submitted with this assessmentResult
+        List<AnswerJpaEntity> allAnswerEntities = answerRepo.findByAssessmentResultId(assessmentResultId);
+
+        /*
+        based on answers, extract all selected options
+        and load all those answerOptions with their impacts
+        */
+        List<Long> allAnswerOptionIds = allAnswerEntities.stream().map(AnswerJpaEntity::getAnswerOptionId).toList();
+
+        Context context = new Context(allQuestionsDto,
+            allAnswerEntities,
+            allQualityAttributeValueEntities,
+            subjectValueEntities,
+            subjectIdToDto,
+            qaIdToWeightMap);
+
+        Map<Long, QualityAttributeValue> qualityAttrIdToValue = buildQualityAttributeValues(context);
```

Comment:
rename variable to `attributeIdToValueMap`
rename method to `buildAttributeValues`